lebedigital/calibration/utils.py

```diff
@@ -0,0 +1,203 @@
+            'path': {'subject': 'ns1:informationbearingentity9', 'predicate': 'ns9:has_text_value'},
+            'file_name': {'subject': 'ns1:informationbearingentity8', 'predicate': 'ns9:has_text_value'}
+        }
+
+        # Path to KG, for testing
+        #path_to_KG = '../../usecases/MinimumWorkingExample/emodul/knowledge_graphs/BA-Losert MI E-Modul 28d v. 04.08.14 Probe 4.ttl'
+
+        path_to_KG = os.path.join(path, exp_name)
+        # initialize the graph
+        knowledge_graph = rdflib.Graph()
+        knowledge_graph.parse(path_to_KG, format='ttl')
+
+        # queries
+        results = query_objects(queries, knowledge_graph)
+
+        # read csv into dataframe
+        file_path = results['path'] + '/' + results['file_name']
+    #skip_last = 145 # hard coded to get the third loading cycle. need somthing better
+    #skip_init = 330
+    if KG == False: # this is just for testing purposes
+        file_path = os.path.join(path,exp_name)
+        results = {}
+        results['length'] = 98.6
+        results['diameter'] = 300.2
+    df = pd.read_csv(file_path, skipfooter=skip_last, engine='python')
+    df = df.drop(labels=range(0, skip_init), axis=0)
+    df['displacement'] = (df['Transducer 1[mm]'] + df['Transducer 2[mm]'] + df['Transducer 3[mm]']) / 3
+    #df['stress'] = df['Force [kN]'] / (np.pi * (float(results['diameter']) / 2) ** 2)
+
+    output = {
+        'exp_name': exp_name,
+        'height': float(results['length']),
+        'diameter': float(results['diameter']),
+        'displacement': np.array(df['displacement']),
+        'force': np.array(df['Force [kN]'])}
+    return output
+
+def load_experimental_data(exp_name, skip_init, skip_last, KG=False, path=None):
+    """
+    Function to interact with the knowledge graph database and get experimental data
+    # TODO : Clarify the units used. Now Force :kN, dispalcement : mm
+    Args:
+        skip_init (int): Initial datavalues to skip as we only need the third loading cycle.
+        skip_last (int): Last datavalues to skip as we only need the third loading cycle.
+        exp_name (string): The experiment name
+
+    Returns:
+        output (dict):
+            The experiment values with keys as 'height' (int), 'diameter' (int), 'displacement' (np.array), 'stress' (np.array)
+    """
+    if KG is True:
+        df = pd.read_csv(emodul_query.input_emodul_data_for_calibration(exp_name)['processedDataPath'],
+                         skipfooter=skip_last)
+    if KG is not True:  # manually passing the processed data path
+        df = pd.read_csv(path, skipfooter=skip_last)
+    df = df.drop(labels=range(0, skip_init), axis=0)
+
+    if KG is True:
+        dia = emodul_query.input_emodul_data_for_calibration(exp_name)['specimenDiameter']
+        height = emodul_query.input_emodul_data_for_calibration(exp_name)['specimenLength']
+    if KG is not True:
+        dia = 98.6
+        height = 300.2
```

Comment:
Height needs to be changed to 100mm as this is the gauge length.

Reply:
This is an old method, not in use as you would have seen.